roadmap/implementers-guide/src/node/utility/candidate-validation.md

```diff
@@ -103,4 +105,24 @@ of a process is less variable under different system conditions. When the
 overall system is under heavy load, the wall clock time of a job is affected
 more than the CPU time.
 
+#### Internal errors
+
+In general, for errors not raising a dispute we have to very careful. This is
+only sound, if we either:
+
+1. Ruled out that error in pre-checking. If something is not checked in
+   pre-checking, even if independent of the candidate we must raise a dispute.
+2. We are 100% confident that it is a hardware/local issue: Like corrupted file,
+   etc.
+
+Reasoning: Otherwise it would be possible to register a PVF where candidates can
+not be checked, but we don't get a dispute - so nobody gets punished. Second, we
```

Comment:
I imagine malicious validators can vote valid and finalize the bad parachain block.

Reply:
I thought the 1/3 threshold was based on all validators, not just the ones that participate. If it's the latter then that's indeed bad. cc @eskimor

Reply:
> I imagine malicious validators can vote valid and finalize the bad parachain block.

appproval-voting should be secure in that case. Any noshow should be covered by more checkers. If the protocol is working as intended, then the only thing that will happen is what @mrcnski wrote: Eventually all honest approval checkers noshow and we have a finality stall, because we don't have enough approvals.

Reply:
With 30 approvals needed and 1/3 malicious by thread model - 200 pravalidators on Kusama, it is totally possible that 30 malicious ones show up.

Reply:
With just one retry and the current timeouts I think this is safe, I was assuming way larger timeouts. So yeah, all good.